crates/oxc_linter/src/rules/jsx_a11y/no_redundant_roles.rs

```diff
@@ -0,0 +1,135 @@
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use phf::{phf_map, phf_set};
+use std::collections::HashMap;
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+    "eslint-plugin-jsx-a11y(no-redundant-roles): The element `{element}` has an implicit role of `{role}`. Defining this explicitly is redundant and should be avoided."
+)]
+#[diagnostic(
+    severity(warning),
+    help("Remove the redundant role `{role}` from the element `{element}`.")
+)]
+struct NoRedundantRolesDiagnostic {
+    #[label]
+    pub span: Span,
+    pub element: String,
+    pub role: String,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoRedundantRoles;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforces that the explicit role property is not the same as implicit/default role property on element.
+    ///
+    /// ### Why is this bad?
+    /// Redundant roles can lead to confusion and verbosity in the codebase.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// <nav role="navigation" />
+    ///
+    /// // Good
+    /// <nav />
+    /// ```
+    NoRedundantRoles,
+    correctness
+);
+
+static DEFAULT_ROLE_EXCEPTIONS: phf::Map<&'static str, phf::Set<&'static str>> = phf_map! {
+    "nav" => phf_set!{"navigation"},
+    "button" => phf_set!{"button"},
+    "body" => phf_set!{"document"},
+};
+
+static EMPTY_SET: phf::Set<&'static str> = phf_set! {};
+
+impl Rule for NoRedundantRoles {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let components: HashMap<String, String> = ctx
+            .settings()
+            .jsx_a11y
+            .components
+            .iter()
+            .map(|(k, v)| (k.to_string(), v.clone()))
+            .collect();
+
+        if let AstKind::JSXOpeningElement(jsx_el) = node.kind() {
+            let element = match &jsx_el.name {
+                JSXElementName::Identifier(identifier) => identifier.name.to_string(),
+                JSXElementName::NamespacedName(namespaced_name) => {
+                    namespaced_name.property.name.to_string()
+                }
+                JSXElementName::MemberExpression(_) => return,
+            };
+
+            let component = components.get(&element).unwrap_or(&element);
+
+            if let Some(JSXAttributeItem::Attribute(attr)) = has_jsx_prop_lowercase(jsx_el, "role")
+            {
+                if let Some(JSXAttributeValue::StringLiteral(role_values)) = &attr.value {
+                    let roles: Vec<String> = role_values
+                        .value
+                        .split_whitespace()
+                        .map(std::string::ToString::to_string)
+                        .collect();
+                    for role in &roles {
+                        let exceptions =
+                            DEFAULT_ROLE_EXCEPTIONS.get(component).unwrap_or(&EMPTY_SET);
```

Comment:
can we either do `unwrap_or_default` or if that doesnt work

let Some(exceptions) = DEFAULT_ROLE_EXCEPTIONS.get(component) else {continue}

or if let if you prefer that

Reply:
Thank you! Indeed.

```rust
let Some(exceptions) = DEFAULT_ROLE_EXCEPTIONS.get(component) else {continue}
```
I was able to remove the `EMPTY_SET` by modifying it using the ↑ method! Thank you 🙇‍♂️ 5fbe14cbfe71f482aafe40b4a51832270e9c4251

Reply:
thanks!